### PR TITLE
Patch reloads live preview on CSS or JS save

### PR DESCRIPTION
**Pull request by K-LV**
WIP. Modified LiveDevMultiBrowser to reload the live preview on CSS and/or JS save. Fix for humphd/brackets#175.

### CODE REVIEW

File: src/LiveDevelopment/LiveDevMultiBrowser.js [hunk 1]
```diff
@@ -241,7 +241,7 @@ define(function (require, exports, module) {
      * @return {string}
      */
     function _resolveUrl(path) {
-        return _server && _server.pathToUrl(path);
+        return _server && path;
```
Comment (humphd) [inline src/LiveDevelopment/LiveDevMultiBrowser.js]: Why this change?

File: src/LiveDevelopment/LiveDevMultiBrowser.js [hunk 2]
```diff
@@ -727,19 +727,19 @@ define(function (require, exports, module) {
...
         // Skip reload if the saved document has live editing enabled
-        if (liveEditingEnabled) {
-            return;
-        }
-        
+        // if (liveEditingEnabled) {
+        //     return;
+        // }
+
         // reload the page if the given document is a JS file related
```
Comment (humphd) [inline src/LiveDevelopment/LiveDevMultiBrowser.js]: Need to update this comment to reflect CSS as well

**Comment by humphd** [comment]
The comment above this function `_onDocumentSaved` needs to be updated to reflect that it's for CSS too, and why Bramble does extra work for reloading CSS.  You can also just remove vs. comment out those lines.

File: src/LiveDevelopment/LiveDevMultiBrowser.js (hunk 1 above)
Comment (K-LV) [inline src/LiveDevelopment/LiveDevMultiBrowser.js]: On line [741] (https://github.com/humphd/brackets/pull/184/files#diff-e31c028c519c0302491877c1d9c0bad1R741), when _liveDocument.isRelated(absolutePath) is called, it returns undefined even for related documents. This is because [isRelated] (https://github.com/humphd/brackets/blob/bramble/src/LiveDevelopment/MultiBrowserImpl/documents/LiveHTMLDocument.js#L299) calls on _resolveUrl() and it returns the Blob URL. However, the related scripts and stylesheets are indexed by filename.

Comment (humphd) [inline src/LiveDevelopment/LiveDevMultiBrowser.js]: What does this change mean for other places where `_resolveUrl` is called?  I'm especially wondering about how the `LiveDoc` ctor uses it?  If you do make this change, it should just be `return path;` without the `_server` check.

File: src/LiveDevelopment/LiveDevMultiBrowser.js [hunk 3]
```diff
@@ -719,6 +719,10 @@ define(function (require, exports, module) {
     /**
      * For files that don't support as-you-type live editing, but are loaded by live HTML documents
      * (e.g. JS files), we want to reload the full document when they're saved.
+     * CSS files do support as-you-type live editing, however, once they are saved, the live
```
Comment (humphd) [inline src/LiveDevelopment/LiveDevMultiBrowser.js]: Add `Bramble note:` to the start of this sentence, so it's clear it's us doing it.

File: src/LiveDevelopment/LiveDevMultiBrowser.js (hunk 1 above)
Comment (K-LV) [inline src/LiveDevelopment/LiveDevMultiBrowser.js]: The only other place I see this being used is in [LiveDevMultiBrowser] (https://github.com/humphd/brackets/blob/bramble/src/LiveDevelopment/LiveDevMultiBrowser.js#L552) when the "ConnectionConnect" event is triggered. However, our code doesn't enter this part because our Transport doesn't trigger this event.

**Comment by humphd** [comment]
@gideonthomas can you review this too, please?

In general, I think this is a sensible change.  I do wonder about the implications of the `_resolveUrl()` change and how it will affect other callers, for example https://github.com/K-LV/brackets/blob/reloadCSSJSOnSave/src/LiveDevelopment/MultiBrowserImpl/documents/LiveHTMLDocument.js#L299.  I'm tempted to say we take this patch and fix bugs with it as we go, if/when we find them.

**Comment by gideonthomas** [comment]
@K-LV can you rebase? Its kinda hard figuring out whats going on in this PR
Also, you said that scripts and stylesheets are indexed by filename...but why does brackets call `urlResolver` in `isRelated`? I assume that it expects to get the blob url (which is what is happening now?). So is it not supposed to call `urlResolver`? Or is the `urlResolver` supposed to do more than just mapping a path to a URL?

**Comment by K-LV** [comment]
I apologize, for the confusion. In Bramble, the scripts and stylesheets are indexed by paths not filenames. `urlResolver` then returns a Blob URL of that CSS or JS. The `isRelated` function fails to find the added CSS/JS becuase it's trying to match the Blob URL with the path. However, in Brackets they are indexed by URLs. Perhaps we can investigate why ours is indexing by path vs URL.
I've also re-based, it's a lot easier to read now.

**Comment by humphd** [comment]
@gideonthomas review ping?  I'd really like to get this landed soon if possible.

**Comment by gideonthomas** [comment]
sorry! @K-LV can you file a follow up bug on that comment and add a comment above the `resolveUrl` method about this. After that r=me

**Comment by K-LV** [comment]
I wrote a bug on the related documents being indexed by path rather than by URL [here] (https://github.com/humphd/brackets/issues/212). I also rebased the PR to the latest.
